goenv/goenv.go

```diff
@@ -249,79 +279,38 @@ func getHomeDir() string {
 	return u.HomeDir
 }
 
-// getGoroot returns an appropriate GOROOT from various sources. If it can't be
-// found, it returns an empty string.
-func getGoroot() string {
-	// An explicitly set GOROOT always has preference.
-	goroot := os.Getenv("GOROOT")
-	if goroot != "" {
-		// Convert to the standard GOROOT being referenced, if it's a TinyGo cache.
-		return getStandardGoroot(goroot)
+// envFromGo gets the value of a variable by querying Go.
+func envFromGo(name string) string {
+	goCmd := os.Getenv("GO")
+	if goCmd == "" {
+		goCmd = "go"
 	}
-
-	// Check for the location of the 'go' binary and base GOROOT on that.
-	binpath, err := exec.LookPath("go")
-	if err == nil {
-		binpath, err = filepath.EvalSymlinks(binpath)
-		if err == nil {
-			goroot := filepath.Dir(filepath.Dir(binpath))
-			if isGoroot(goroot) {
-				return goroot
-			}
-		}
+	cmd := exec.Command(goCmd, "env", name)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return ""
 	}
 
-	// Check what GOROOT was at compile time.
-	if isGoroot(runtime.GOROOT()) {
-		return runtime.GOROOT()
-	}
+	return strings.TrimSpace(out.String())
+}
 
-	// Check for some standard locations, as a last resort.
-	var candidates []string
-	switch runtime.GOOS {
-	case "linux":
-		candidates = []string{
-			"/usr/local/go",     // manually installed
-			"/usr/lib/go",       // from the distribution
-			"/snap/go/current/", // installed using snap
-		}
```

Comment:
This removes functionality. Is this intended?
Normally this is used to detect a Go installation when the `go` binary is not in `$PATH`.

Reply:
The `go list` command already does not use this autodetection, so if that was intended, it does not work.

Reply:
The `/usr/local/go` kinda makes sense I guess, but the installation instructions describe adding it to path, so I don't think that makes too much sense here. As for the other two, they are automatically in `$PATH`.